src/pages/consultations/EditConsultationPage.tsx

```diff
@@ -1,137 +1,180 @@
+    createConsultation(formData, {
+      onSuccess: (consultation) => {
+        toast({ title: 'Konzultáció sikeresen létrehozva!', status: 'success' })
+        navigate(`/consultations/${consultation.id}`)
+      },
+      onError: (e: KonziError) => {
+        toast(generateToastParams(e))
+      }
+    })
   }
 
-  useEffect(() => {
-    setTimeout(() => {
-      const c = testConsultationDetails.find((c) => c.id === consultationId)
-      if (!c) {
-        setLoading(false)
-        setConsultation(undefined)
-        return
+  const editCons = (formData: CreateConsultation) => {
+    updateConsultation(formData, {
+      onSuccess: (consultation) => {
+        toast({ title: 'Konzultáció sikeresen módosítva!', status: 'success' })
+        navigate(`/consultations/${consultation.id}`)
+      },
+      onError: (e: KonziError) => {
+        toast(generateToastParams(e))
       }
-      setConsultation(c)
-      setName(c.name)
-      setLocation(c.location)
-      setDescription(c.descMarkdown)
-      setStartDate(c.startDate)
-      setEndDate(c.endDate)
-      setSubject(c.subject)
-      setPresentations(c.presentations)
-      setTargetGroups(c.targetGroups)
-      setLoading(false)
-    }, 1000)
-  }, [])
+    })
+  }
+
+  const form = useForm<CreateConsultationForm>({
+    defaultValues: consultation
+      ? {
+          name: consultation.name,
+          location: consultation.location,
+          descMarkdown: consultation.descMarkdown,
+          startDate: new Date(consultation.startDate),
+          endDate: new Date(consultation.endDate),
+          subject: consultation.subject,
+          presenters: consultation.presentations,
+          targetGroups: consultation.targetGroups
+        }
+      : { targetGroups: [], presenters: [], startDate: new Date(), endDate: new Date() },
+    mode: 'all'
+  })
+
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    watch,
+    formState: { errors, isValid, isSubmitted }
+  } = form
+
+  const onSubmit = handleSubmit((data) => {
+    const formData = {
+      name: data.name,
+      location: data.location,
+      startDate: new Date(data.startDate),
+      endDate: new Date(data.endDate),
+      descMarkdown: data.descMarkdown,
+      subjectId: data.subject.id,
+      presenterIds: data.presenters.map((p) => p.id),
+      targetGroupIds: data.targetGroups.map((g) => g.id)
+    }
+    newConsultation ? createCons(formData) : editCons(formData)
+  })
+
+  useEffect(() => {
+    if (consultation && !watch('name')) {
+      setValue('name', consultation.name)
+      setValue('location', consultation.location)
+      setValue('descMarkdown', consultation.descMarkdown)
+      setValue('presenters', consultation.presentations)
+      setValue('subject', consultation.subject)
+      setValue('targetGroups', consultation.targetGroups)
+      setValue('startDate', new Date(consultation.startDate))
+      setValue('endDate', new Date(consultation.endDate))
```

Comment:
Isn't it a form reset?

Reply:
Doesn't work, unfortunately. The default values are defined only if the query's result was cached, otherwise it has to request the data from the server and set it here. Could be solved by not using react query, and supplying the default value in our own async function, but the caching mechanism is actually useful here: if the user clicks on the edit button on the consultation details page, the consultation data is cached and it doesn't even have to send a request to the server. This part is only needed if the user refreshes the edit form, for example.